services/src/main/java/com/linkedin/kmf/common/TopicFactory.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2016 LinkedIn Corp. Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.linkedin.kmf.common;
+
+import java.util.Map;
+
+
+/**
+ * Constructs the monitored topic if it does not exist.
+ *
+ * Implementations of this class should have a public, no-arg constructor.
```

Comment:
Kafka seems to follow the pattern of having these plugin classes have a no-arg constructor and method like void configure(Map<String, ?>).  I do like just being able to call newInstance() or the config class's equivalent method.  It also decouples the configuration of a class from it's initialization which might be important if "final" methods need to be assigned some value, but the class needs to be configured later.  I don't see a case where that might happen here.  The other approach ( Constructor(Map<String, ?> ) ), has the advantage of constructing valid objects which is something constructors should enforce when possible.   I don't know what do you think?

In order to construct the ACL an internal implementation of TopicFactoryneeds:
1) Topic Name
2) Principals (which can be extracted from the identity file)
3) Cluster Type
4) Fabric type
5) The datavault uri

Some of these properties would already be present in the TopicManagementService and in the produce service or the consume service configuration.